### ISSUE TITLE
Marshmallow plugin doesn't document the format of fields.DateTime() correctly

### ISSUE THREAD

**Issue by TheBigRoomXXL** [marshmallow-code/apispec#814]
Marshmallow support multiple format for DateTime: 

> **format:** Either "rfc" (for RFC822), "iso" (for ISO8601), "timestamp", "timestamp_ms" (for a POSIX timestamp) or a date format string. If [None](https://python.readthedocs.io/en/latest/library/constants.html#None), defaults to “iso”.

I was expecting apispec to adapt documentation base on the format but apispec always default to iso format. To change this behavior ,  I guess  the `FieldConverterMixin.field2type_and_format` would need to be changed but I don't know exactly how. 

I don't know if that qualify as a "bug" but it would certainly be nice to have that in the future.

**Comment by lafrech** [comment]
Yes, in OpenAPI "date-time" implies iso and other formats should use a specific pattern.

https://stackoverflow.com/a/49379235

Let's label this a bug.

This deserves a new  `field2datetime` function in field_converter.py.

Help on this welcome.

Hopefully the correct regex for other formats can be found somewhere.

Users using custom format would have to define the regex themselves.

**Comment by TheBigRoomXXL** [comment]
> Yes, in OpenAPI "date-time" implies iso and other formats should use a specific pattern.

Yes but timestamp is a little different than custom date because it should have the type `"integer"` instead of `"string"`. 
Also OpenAPI format is an open field so I think we should also add `'timestamp'` or  `'timestamp_ms'` as format .

> Hopefully the correct regex for other formats can be found somewhere.

I couldn't find any regex for RFC822 so I did my own: 
```regex
^((Mon|Tue|Wed|Thu|Fri|Sat|Sun), ){0,1}\d{2} (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) \d{4} \d{2}:\d{2}:\d{2} (UT|GMT|EST|EDT|CST|CDT|MST|MDT|PST|PDT|(Z|A|M|N)|(\+|-)\d{4})$
```
Some example of it here: [https://regex101.com/r/hOctM6/1](https://regex101.com/r/hOctM6/1)

It's not perfect. Some DateTime string not conform with RFC822 will slip through but it's very descriptive and relatively short.. 

> Users using custom format would have to define the regex themselves.

Yes. I think a `pattern` parameter inside the metadata of field is a good place to put it. 

> Help on this welcome.

Here is my pull request : https://github.com/marshmallow-code/apispec/pull/815. I hope it's good with you!

**Comment by sloria** [comment]
resolved by #815 and released in 6.4.0
